Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1101,6 +1101,22 @@ public extension Purchases {
 
 #endif
 
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    func handleObserverModeTransaction(
+        purchaseResult: StoreKit.Product.PurchaseResult
+    ) async throws -> StoreTransaction? {
+        guard !self.systemInfo.observerMode else {
+            throw NewErrorUtils.observerModeNotEnabledError()
+        }
+        let (userCancelled, transaction) = try await self.purchasesOrchestrator.storeKit2TransactionListener.handle(
+            purchaseResult: purchaseResult, fromTransactionUpdate: true)
+
+        if userCancelled, self.systemInfo.dangerousSettings.customEntitlementComputation {
```

Comment:
I'm not entirely sure about the `customEntitlementComputation` check here. This snippet was taken from `PurchasesOrchestrator`:

https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift#L501-L502

Reply:
Removed it for now